IronRoadForChildren/Features/Program/ProgramViewModel.swift

```diff
@@ -7,25 +7,115 @@
 
 import Foundation
 import Networking
+import SwiftUI
 
 @MainActor
 class ProgramViewModel: ObservableObject {
 	@Published var selectedTab = 0
 
-	@Published var events: [Event] = []
+	private var allEvents: [Event] = []
+
 	@Published var isLoadingEvents = false
 
-	@Published var error: String = ""
+	@Published var eventDays: [EventDay] = []
+	@Published var eventCategories: [EventCategory] = []
+	@Published var filteredCategorie: EventCategory? = nil
 
-	init() {}
+	@Published var error: String? = nil
 
-	private func fetchEvents() async throws {
-		isLoadingEvents = true
+	init() {
+		loadEvents()
+	}
+
+	func loadEvents() {
+		Task.detached { @MainActor in
+			do {
+				withAnimation {
+					self.isLoadingEvents = true
+				}
+
+				self.error = nil
+				try await self.fetchCategories()
+				try await self.fetchEvents()
+				await self.separateEventToDays()
+
+				withAnimation {
+					self.isLoadingEvents = false
+				}
+			} catch {
+				withAnimation {
+					self.isLoadingEvents = false
+					self.error = error.localizedDescription
+				}
+			}
+		}
+	}
 
+	private func fetchEvents() async throws {
 		let url = world.serverUrlWith(path: "/api/events")
 		let (body, _) = try await URLSession.shared.dataArray(.get, from: url, responseType: Event.self)
 
-		events = body
-		isLoadingEvents = false
+		allEvents = body
+	}
+
+	private func fetchCategories() async throws {
+		let url = world.serverUrlWith(path: "/api/eventCategories")
+		let (body, _) = try await URLSession.shared.dataArray(.get, from: url, responseType: EventCategory.self)
+
+		eventCategories = body
+	}
+
+	private func separateEventToDays() async {
+		let allEvents = allEvents
+
+		guard !allEvents.isEmpty else {
+			withAnimation {
+				self.eventDays = []
+			}
+			return
+		}
+
+		var eventsByDays: [String: [Event]] = [:]
+
+		allEvents.forEach { event in
+			let date = world.localDate(of: event.startDateTimeInUTC)
+
+			if eventsByDays.contains(where: { $0.key == date }) {
+				eventsByDays[date]?.append(event)
+			} else {
+				let newEvents = [event]
+				eventsByDays[date] = newEvents
+			}
```

Comment:
this adds the events in "random" order to the day. I think they should be sorted by startTime